californium-core/src/main/java/org/eclipse/californium/core/network/Matcher.java

```diff
@@ -104,6 +105,9 @@ public Matcher(NetworkConfig config) {
 		
 		healthStatusLevel = Level.parse(config.getString(NetworkConfig.Keys.HEALTH_STATUS_PRINT_LEVEL));
 		healthStatusInterval = config.getInt(NetworkConfig.Keys.HEALTH_STATUS_INTERVAL);
+		multicastMaxLifetime = config.getInt(NetworkConfig.Keys.NON_LIFETIME) 
+				+ config.getInt(NetworkConfig.Keys.MAX_LATENCY) 
+				+ config.getInt(NetworkConfig.Keys.MAX_SERVER_RESPONSE_DELAY);
 	}
```

Comment:
According RFC7252, 4.8.2, page 30, the `NON_LIFETIME` seems to already include the `MAX_LATENCY`.
RFC7252, 8.2, page 66, the multicastMaxLifetime considers also the `LEISURE` time.

Reply:
According RFC 7390, 2.5, page 10, the time between reuse of Token values
   used in multicast requests MUST be greater than: `NON_LIFETIME + MAX_LATENCY + MAX_SERVER_RESPONSE_DELAY`.

Reply:
Reading that RFC7390 it confuses me. I don't know, why the calculation there include the MAX_LATENCY twice (explicit and implicit with NON_LIFETIME).